suitcase/pyxrf/export.py

```diff
@@ -0,0 +1,414 @@
+    create_each_det : Bool, optional
+        if number of point is too large, only sum data is saved in h5 file
+    """
+    interpath = 'xrfmap'
+    sum_data = None
+
+    with h5py.File(fpath, 'a') as f:
+        if create_each_det:
+            for n in range(num_det):
+                detname = 'det' + str(n+1)
+                new_data = data[detname]
+
+                if sum_data is None:
+                    sum_data = new_data
+                else:
+                    sum_data += new_data
+
+                dataGrp = f.create_group(interpath+'/'+detname)
+                ds_data = dataGrp.create_dataset('counts', data=new_data,
+                                                 compression='gzip')
+                ds_data.attrs['comments'] = (
+                    'Experimental data from channel {}'.format(n))
+
+        else:
+            sum_data = data['det_sum']
+
+        # summed data
+        if sum_data is not None:
+            dataGrp = f.create_group(interpath+'/detsum')
+            ds_data = dataGrp.create_dataset('counts', data=sum_data,
+                                             compression='gzip')
+            ds_data.attrs['comments'] = 'Experimental data from channel sum'
+
+        # add positions
+        if 'pos_names' in data:
+            dataGrp = f.create_group(interpath+'/positions')
+            pos_names = data['pos_names']
+            pos_data = data['pos_data']
+            dataGrp.create_dataset('name', data=helper_encode_list(pos_names))
+            dataGrp.create_dataset('pos', data=pos_data)
+
+        # scaler data
+        if 'scaler_data' in data:
+            dataGrp = f.create_group(interpath+'/scalers')
+            scaler_names = data['scaler_names']
+            scaler_data = data['scaler_data']
+            dataGrp.create_dataset('name',
+                                   data=helper_encode_list(scaler_names))
+            dataGrp.create_dataset('val', data=scaler_data)
+
+
+def helper_encode_list(data, data_type='utf-8'):
+    return [d.encode(data_type) for d in data]
+
+
+def helper_scaler_value_2D(name_list, data, datashape):
+    """
+    Helper function to transfer 1D scalar data into 2D with given shape.
+
+    Parameters
+    ----------
+    name_list : list
+        list of scalar names
+    data : dictionary or pd.series
+        1D scalar data
+    datashape : list or tuple
+        shape in 2D
+
+    Returns
+    -------
+    combined scalar data in 3D
+    """
+    pos_data = np.zeros([datashape[0], datashape[1], len(name_list)])
+    for i, v in enumerate(name_list):
+        # keep shape unchanged, so stopped/aborted run can be handled.
+        posv = np.zeros(datashape[0]*datashape[1])
+        data[v] = np.asarray(data[v])
+        posv[:data[v].shape[0]] = np.asarray(data[v])  # scan may not finish.
+        pos_data[:, :, i] = posv.reshape([datashape[0], datashape[1]])
+    return pos_data
+
+
+def flip_data(input_data, subscan_dims=None):
```

Comment:
My initial reaction to this is that analysis/processing code should stay out of this I/O library. Is there another place we can put this?